variant_qc/exomes_genomes_coverage.py

```diff
@@ -0,0 +1,145 @@
+from gnomad_hail import *
+from gnomad_hail.resources.variant_qc import get_ucsc_mappability
+
+def import_cds_from_gtf() -> hl.Table:
+    """
+    Creates a HT with a row for each base / gene that is in a CDS in gencode v19
+    :return: HT
+    :rtype: Table
+    """
+    gtf = hl.experimental.import_gtf(
+        'gs://hail-common/references/gencode/gencode.v19.annotation.gtf.bgz',
+        reference_genome='GRCh37',
+        skip_invalid_contigs=True, min_partitions=200
+    )
+    gtf = gtf.filter((gtf.feature == 'CDS') & (gtf.transcript_type == 'protein_coding') & (gtf.tag == 'basic'))
+    gtf = gtf.annotate(locus=hl.range(gtf.interval.start.position, gtf.interval.end.position).map(lambda x: hl.locus(gtf.interval.start.contig, x, 'GRCh37')))
+    gtf = gtf.key_by().select('gene_id', 'locus', 'gene_name').explode('locus')
+    gtf = gtf.key_by('locus', 'gene_id').distinct()
+    return gtf.checkpoint('gs://gnomad-tmp/gencode_grch37.gene_by_base.ht', overwrite=True)
+
+def compute_per_base_cds_coverage():
+    """
+    Creates a HT with a row for each base and gene that is in a CDS in gencode v19 with the following information:
+    1) gene id
+    2) gnomAD exomes coverage by platform
+    3) gnomAD genomes coverage by PCR status
+    """
+
+    def get_haploid_coverage_struct(mt):
+        return hl.cond(
+            mt.locus.in_autosome_or_par(),
+            hl.struct(
+                mean=0.5*hl.agg.sum(mt.mean  * mt.n)/hl.agg.sum(mt.n),
+                median=0.5*hl.median(hl.agg.collect(mt.median)),
+                **{
+                    f'over_{i/2}': hl.agg.sum(mt[f'over_{float(i)}']) / hl.agg.sum(mt.n)
```

Comment:
Seems mildly dangerous to be using a fractional float in the string (might have been easier, if more awkward, to multiply the haploid ones by 2 rather than divide diploid), but I guess if it works it works

Reply:
Yeah, the problem here is that we didn't compute the right numbers from the BAMs.

Reply:
Yeah agreed, I just mean the practical purpose of a float in a string (in case you get something like `over_10.00000` but I think python doesn't do this, so it's fine)